events/importer/helper/importers/vapaaehtoistyofi/record.py

```diff
@@ -0,0 +1,135 @@
+            else:
+                self.no_time = False
+            self.description = json_dict['description']
+            self.contact_details = json_dict['contactDetails']
+            self.themes = json_dict['id']
+            self.timestamp_publish = datetime.datetime.utcfromtimestamp(json_dict['timeStampPublicationDate'])
+            # self.publicationTime = json_dict['id']
+            if json_dict['status'] in self.STATUS:
+                self.status = json_dict['id']
+            else:
+                raise RuntimeError("Unknown status %d!" % json_dict['status'])
+            self.creator_id = json_dict['creator']
+            if json_dict['timeStampAdded']:
+                self.timestamp_inserted = datetime.datetime.utcfromtimestamp(json_dict['timeStampAdded'])
+            else:
+                self.timestamp_inserted = None
+            if json_dict['timeStampLastUpdated']:
+                self.timestamp_updated = datetime.datetime.utcfromtimestamp(json_dict['timeStampLastUpdated'])
+            else:
+                self.timestamp_updated = None
+        else:
+            self.id = None
+            self.organization_id = None
+            self.organization_name = None
+            self.title = None
+            self.address = None
+            self.address_coordinates = None
+            self.tags = []
+
+            self.no_time = None
+            self.timestamp_start = None
+            self.timestamp_end = None
+
+            self.description = None
+            self.contact_details = None
+            self.themes = None
+            self.timestamp_publish = None
+
+            self.status = None
+            self.creator_id = None
+            self.timestamp_inserted = None
+            self.timestamp_updated = None
+
+    def get_url_locale(self, locale):
+        if locale not in self.LOCALE:
+            raise ValueError("Unknown locale '%s'!" % locale)
+        return self.TASK_URL_FORMAT % (locale, self.id)
+
+    def __copy__(self) -> object:
+        newrecord = Record(None)
+        newrecord.id = self.id
+        newrecord.organization_id = self.organization_id
+        newrecord.organization_name = self.organization_name
+        newrecord.title = self.title
+        newrecord.address = self.address
+        newrecord.address_coordinates = self.address_coordinates
+        newrecord.tags = self.tags
+
+        newrecord.no_time = self.no_time
+        newrecord.timestamp_start = self.timestamp_start
+        newrecord.timestamp_end = self.timestamp_end
+
+        newrecord.description = self.description
+        newrecord.contact_details = self.contact_details
+        newrecord.themes = self.themes
+        newrecord.timestamp_publish = self.timestamp_publish
+
+        newrecord.status = self.status
+        newrecord.creator_id = self.creator_id
+        newrecord.timestamp_inserted = self.timestamp_inserted
+        newrecord.timestamp_updated = self.timestamp_updated
```

Comment:
Why is this needed? Doesn't `self.__copy__()` already do the same?

```
In [50]: r1.__class__
Out[50]: events.importer.helper.importers.vapaaehtoistyofi.record.Record

In [51]: r1.__dict__
Out[51]:
{'id': 'id',
 'organization_id': 'organization_id',
 'organization_name': 'organization_name',
 'title': 'title',
 'address': 'address',
 'address_coordinates': 'address_coordinates',
 'tags': [],
 'no_time': None,
 'timestamp_start': None,
 'timestamp_end': None,
 'description': None,
 'contact_details': None,
 'themes': None,
 'timestamp_publish': None,
 'status': None,
 'creator_id': None,
 'timestamp_inserted': None,
 'timestamp_updated': None}

In [57]: r2 = r1.__copy__()

In [58]: r2.__dict__ == r1.__dict__
Out[58]: True
```

I could have tried if there would be unit tests for this. ;)

Reply:
My Python:
```
AttributeError: 'Record' object has no attribute '__copy__'
```

Reply:
This one solved by dropping `__copy()`.